gatsby/plugins/gatsby-source-cks/src/node-creation/__tests__/changes.test.ts

```diff
@@ -0,0 +1,84 @@
+import { SourceNodesArgs } from "gatsby";
+
+import { createChangeNodes } from "../changes";
+import { ApiTopicChangeResponse } from "../../api/types";
+
+describe("createChangeNodes", () => {
+	const createNodeId = jest.fn(),
+		createContentDigest = jest.fn(),
+		createNode = jest.fn();
+
+	const sourceNodesArgs: SourceNodesArgs = ({
+		createNodeId,
+		createContentDigest,
+		actions: {
+			createNode,
+		},
+	} as unknown) as SourceNodesArgs;
```

Comment:
interesting syntax

Reply:
Yeah downside of TS sometimes needing to cast like that when it doesn't think objects are the same, or when you're mocking and you only need a few properties. Otherwise you'd have to add every single properties for each mocked type.